### PR TITLE
Search MVP in one commit

### PR DESCRIPTION
**Pull request by bryan-gilbert**
Created by deleting my fork. Recreate fork.  Take a directory with all the code and remove the .git references.  Copy over the new fork.  Then carefully determine what is in and not in.

### CODE REVIEW

File: modules/core/server/controllers/core.dbmodel.controller.js [hunk 1]
```diff
@@ -1053,18 +1048,16 @@ _.extend (DBModel.prototype, {
...
 								console.log('search.count.error = ' + JSON.stringify(e));
 								self.complete(reject, 'search');
 							} else {
-								// console.log('search.count.completed. total = ', c);
 								resolve({data: data, count: c});
 							}
 						});
 					} else {
-						// console.log('search.error = ' + JSON.stringify(error));
+						console.log('search.error = ' + JSON.stringify(error));
```
Comment (marklise) [inline modules/core/server/controllers/core.dbmodel.controller.js]: Remove console.log

File: modules/core/server/controllers/core.dbmodel.controller.js [hunk 2]
```diff
@@ -1076,7 +1069,55 @@ _.extend (DBModel.prototype, {
...
+						return -1 * SearchUtil.relevanceRanking(item, terms.searchTerms, searchFields);
+					});
+
+					// paginate ...
+					sorted = sorted.slice(skip, skip + limit);
+
+					// return the paginated results with the total count
+					resolve({data: sorted, count: cnt});
+				} else {
+					console.log('search.error = ' + JSON.stringify(error));
```
Comment (marklise) [inline modules/core/server/controllers/core.dbmodel.controller.js]: Remove console.log

File: modules/core/server/controllers/core.routes.controller.js [hunk 3]
```diff
@@ -241,6 +241,30 @@ var setModel = function (Dbclass, name) {
 	};
 };
 exports.setModel = setModel;
+
+var setModelByQuery = function () {
+	return function (req, res, next) {
+		var name = req.query.collection;
+		var Dbclass;
+		var fields;
+		switch(name) {
```
Comment (marklise) [inline modules/core/server/controllers/core.routes.controller.js]: Declare Dbclass at the top.
Remove the switch statement, turn it into an IF.
Reject if the name != 'documents'

File: modules/core/server/controllers/core.routes.controller.js [hunk 4]
```diff
@@ -275,6 +299,17 @@ var setAndRun = function (Dbclass, f) {
 exports.setAndRun = setAndRun;
 
 
+function searchMiddle(req, res, next) {
+	function workerFunction (model, searchFields, queryOptions) {
+		return model.search (searchFields, queryOptions);
+	}
+	setSessionContext (req)
+		.then (function (opts) {
```
Comment (marklise) [inline modules/core/server/controllers/core.routes.controller.js]: indentation

File: modules/core/server/controllers/core.routes.controller.js [hunk 5]
```diff
@@ -457,4 +449,49 @@ exports.setCRUDRoutes = function (app, basename, DBClass, policy, which, policym
...
 
+	if (options) {
+		if (options.filterBy) {
+			query = options.filterBy;
+		}
+		if (options.filterByFields) {
+			for (var key in options.filterByFields) {
+				if (options.filterByFields.hasOwnProperty(key)) {
+					filter[key] = options.filterByFields[key];
+					console.log(filter);
```
Comment (marklise) [inline modules/core/server/controllers/core.routes.controller.js]: Remove console.log

File: modules/core/server/routes/core.server.routes.js [hunk 6]
```diff
@@ -1,10 +1,20 @@
 'use strict';
 
 module.exports = function (app) {
-  // Root routing
-  var core = require('../controllers/core.server.controller');
+	var path    = require('path');
```
Comment (marklise) [inline modules/core/server/routes/core.server.routes.js]: A mix of spaces and tabs between vars and "="

File: modules/documents/client/views/document-manager-info-panel.html [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+				<button class="btn btn-default close" type="button" ng-click="infoPanel.close()">
+					<span aria-hidden="true">&times;</span>
+				</button>
+				<h2 title="{{infoPanel.data.displayName | removeExtension}}">{{infoPanel.data.displayName | removeExtension }}</h2>
+			</div>
+			<div class="title-panel-btns">
+				<a ng-if="infoPanel.data.userCan.read"
+					 class="btn btn-sm btn-default"
+					 href="/api/document/{{ infoPanel.data._id }}/fetch"
+					 target="_self"
```
Comment (marklise) [inline modules/documents/client/views/document-manager-info-panel.html]: target _blank.

File: modules/search/client/directives/search.directive.js [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+		link: function (scope, element, attrs) {
+			element.on('click', function () {
+				var modalInstructions = $modal.open({
+					animation: true,
+					templateUrl: 'modules/search/client/views/partials/modal-search-instructions.html',
+					controller: 'controllerModalSearchInstructions',
+					controllerAs: 'instruct',
+					size: 'lg'
+				});
+				modalInstructions.result.then(function (data) {
```
Comment (marklise) [inline modules/search/client/directives/search.directive.js]: What's the purpose of this?

**Comment by bryan-gilbert** [comment]
ready

**Pull request merged**